src/main/java/com/cloudbees/jenkins/support/impl/SlaveLaunchLogs.java

```diff
@@ -73,128 +86,80 @@ public String getDisplayName() {
-            for (File f : files) {
-                container.add(new LaunchLogsFileContent(
-                        "nodes/slave/{0}/launchLogs/{1}",
-                        new String[] {agent.getName(), f.getName()}, f, FileListCapComponent.MAX_FILE_SIZE));
-            }
         }
     }
 
-    static class Agent implements Comparable<Agent> {
-        /**
-         * Launch log directory of the agent: logs/slaves/NAME
-         */
-        final File dir;
-
-        final long time;
-
-        Agent(File dir, File lastLog) {
-            this.dir = dir;
-            this.time = lastLog.lastModified();
-        }
+    @Extension
+    public static final class LogArchiver extends ConsoleLogFilter {
 
-        /** Agent name */
-        String getName() {
-            return dir.getName();
-        }
+        final File log = new File(SafeTimerTask.getLogsRoot(), "nodeLaunchLogs.txt");
+        private final RewindableRotatingFileOutputStream stream =
+                new RewindableRotatingFileOutputStream(log, MAX_ROTATE_LOGS);
 
-        /**
-         * Use the primary log file's timestamp to compare newer agents from older agents.
-         *
-         * sort in descending order; newer ones first.
-         */
-        public int compareTo(Agent that) {
-            return Long.compare(that.time, this.time);
+        @Override
+        public OutputStream decorateLogger(Computer computer, OutputStream logger) {
+            if (computer instanceof SlaveComputer) {
+                return new TeeOutputStream(logger, new PrefixedStream(stream, computer.getName()));
+            } else {
+                return logger;
+            }
         }
 
+        @SuppressWarnings("rawtypes")
         @Override
-        public boolean equals(Object o) {
-            if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
+        public OutputStream decorateLogger(Run build, OutputStream logger) throws IOException, InterruptedException {
+            return logger;
+        }
+    }
 
-            Agent agent = (Agent) o;
+    static class PrefixedStream extends LineTransformationOutputStream.Delegating {
+        private final String name;
 
-            return time == agent.time;
+        PrefixedStream(OutputStream out, String name) {
+            super(out);
+            this.name = name;
         }
 
         @Override
-        public int hashCode() {
-            return (int) (time ^ (time >>> 32));
-        }
-
-        /**
-         * If the file is more than 7 days old, it is considered too old.
-         */
-        public boolean isTooOld() {
-            return time < System.currentTimeMillis() - TimeUnit.DAYS.toMillis(7);
+        protected void eol(byte[] b, int len) throws IOException {
+            synchronized (out) {
+                out.write('[');
+                out.write(DateTimeFormatter.ISO_INSTANT
+                        .format(Instant.now().truncatedTo(ChronoUnit.MILLIS))
```

Comment:
See https://github.com/jenkinsci/lib-support-log-formatter/issues/64 for other logs.